clustering.py

```diff
@@ -42,6 +43,47 @@ def visualize_clusters(df):
     plt.show()
 
 
+
+
+
+def cluster_analysis(df):
+
+    pca_df = add_pca_features(df)
+
+    pca_mat = pca_df.drop(columns=["genre"])
+
+
+    kmeans = KMeans(
+    n_clusters=8, init="random", max_iter=300, n_init=10, random_state=29
+    )
+    kmeans.fit(pca_mat)
+    pca_df["cluster"] = kmeans.labels_
```

Comment:
This should not be here. The function should take as argument a dataframe that already contains the clusters. Run `create_clustered_df()` to have this dataframe from the pca dataframe.